modules/nlu/src/backend/application/index.ts

```diff
@@ -0,0 +1,129 @@
+import * as NLU from 'common/nlu/engine'
+import _ from 'lodash'
+import yn from 'yn'
+
+import { IBotFactory } from './bot-factory'
+import { IBotService } from './bot-service'
+import { BotNotMountedError } from './errors'
+import { ITrainingQueue } from './training-queue'
+import { ITrainingRepository } from './training-repo'
+import { Predictor, BotConfig, TrainingSession, TrainingState, TrainingId } from './typings'
+
+export class NLUApplication {
+  private _queueTrainingOnBotMount: boolean
+
+  constructor(
+    private _trainingQueue: ITrainingQueue,
+    private _engine: NLU.Engine,
+    private _botFactory: IBotFactory,
+    private _botService: IBotService,
+    queueTrainingOnBotMount: boolean = true
+  ) {
+    this._queueTrainingOnBotMount = queueTrainingOnBotMount
+  }
+
+  public async initialize() {
+    await this._trainingQueue.initialize()
+  }
+
+  public get trainRepository(): ITrainingRepository {
+    return this._trainingQueue.repository
+  }
+
+  public teardown = async () => {
+    for (const botId of this._botService.getIds()) {
+      await this.unmountBot(botId)
+    }
+    return this._trainingQueue.teardown()
+  }
+
+  public getHealth() {
+    return this._engine.getHealth()
+  }
+
+  public async getTraining(botId: string, language: string): Promise<TrainingState> {
+    return this._trainingQueue.getTraining({ botId, language })
+  }
+
+  async getAllTrainings(): Promise<TrainingSession[]> {
+    return this._trainingQueue.getAllTrainings()
+  }
+
+  async pauseTrainings() {
```

Comment:
used somewhere ?

Reply:
I believe its used in testing infrastructure, but I'm not sure. Will check